File: Data-Structures-And-Libraries/Python/URI-1260.py
def get_species(dictionary: "dict of species", flag: "indicate calling scope") -> float:
    trees = [specie for specie in dictionary.keys()]
    trees.sort()
    total = sum(list(dictionary.values()))
    for tree in trees:
        print("%s %.4f" % (tree, (dictionary[tree] / total) * 100))
    dictionary.clear()
    if flag is "loop":
        print()


def main():
    cases = int(input())
    line_feed = input()
    total_feed = 1
    trees_dict = {}
    while True:
        try:
            specie = input()
            if len(specie) == 0:
                total_feed += 1
                get_species(trees_dict, "loop")
            else:
                try:
                    trees_dict[specie] += 1
                except KeyError:
                    trees_dict[specie] = 1
        except EOFError:
            get_species(trees_dict, "end")
            break


main()

File: Data-Structures-And-Libraries/Java/URI-1260.java
import java.io.IOException;
import java.io.File;
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.Locale;
import java.util.Scanner;
 
public class Main {
 
    public static void main(String[] args) throws IOException {
 
        Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int cases = sc.nextInt();
		sc.nextLine(); /*consuming the next empty line */
		sc.nextLine();
		
		HashMap<String, Integer> treesMap = new HashMap<String, Integer>();
		int totalTrees = 0;
		
		while(sc.hasNextLine()) {
			String specie = sc.nextLine();
			
			if(specie.length() == 0) {
				getSpecies(treesMap, totalTrees, "loop");
				totalTrees = 0;
				treesMap.clear();
				
			}
			
			else {
				
				if(treesMap.get(specie) != null) {
					treesMap.replace(specie, treesMap.get(specie) + 1);
				}
				else {
					treesMap.put(specie, 1);	
			}
				totalTrees += 1;
		  }		
		}	
		getSpecies(treesMap, totalTrees, "end");
		sc.close();
 
    }
    
    static void getSpecies(HashMap<String, Integer> dictionary,int totalTrees, String flag) {
		ArrayList<String> species = new ArrayList<String>();
		for(String specie : dictionary.keySet()) {
			species.add(specie);		
		}
		Collections.sort(species);
		
		for(String specie: species) {
			System.out.printf("%s %.4f%n", specie, (float) dictionary.get(specie) / totalTrees * 100);		
		}
		dictionary.clear();
		totalTrees = 0;
		
		if(flag == "loop") {
			System.out.println();			
		}
	}
 
}
